src/sonic-config-engine/sonic_device_util.py

```diff
@@ -73,10 +108,14 @@ def get_system_mac():
         # Try valid mac in eeprom, else fetch it from eth0
         platform = get_platform_info(get_machine_info())
         hwsku = get_machine_info()['onie_machine']
-        profile_cmd = 'cat /usr/share/sonic/device/' + platform +'/'+ hwsku +'/profile.ini | cut -f2 -d='
+        profile_cmd = 'cat' + SONIC_DEVICE_PATH + '/' + platform +'/'+ hwsku +'/profile.ini | cut -f2 -d='
         hw_mac_entry_cmds = [ profile_cmd, "sudo decode-syseeprom -m", "ip link show eth0 | grep ether | awk '{print $2}'" ]
     else:
-        hw_mac_entry_cmds = [ "ip link show eth0 | grep ether | awk '{print $2}'" ]
+        if namespace is not None:
+            ip_link_show_cmd = "sudo ip -n {} link show eth0".format(namespace)
+        else:
+            ip_link_show_cmd = "ip link show eth0"
+        hw_mac_entry_cmds = [ip_link_show_cmd + " | grep ether | awk '{print $2}'"]
```

Comment:
is it possible to read all this info from /sys?

Reply:
Fixed in latest commit